src/libopensc/card-iasecc.c

```diff
@@ -916,7 +969,9 @@ iasecc_select_file(struct sc_card *card, const struct sc_path *path,
 
 	sc_print_cache(card);
 
-	if (card->cache.valid && card->cache.current_df && lpath.type == SC_PATH_TYPE_DF_NAME
+	if ((card->type != SC_CARD_TYPE_IASECC_CPX) &&
+	    (card->type != SC_CARD_TYPE_IASECC_CPXCL)  &&
```

Comment:
This is already second time you have the same condition. I think it would be a great candidate for some helper function `inline static int iasecc_is_cpx(card)` to make the changes here less intrusive.

Reply:
Thanks, I factorized it with a specific commit in order to track the updates. https://github.com/OpenSC/OpenSC/pull/2217/commits/656d203e195ef6216f9d00cc264fb0dcf1faeaa4